File: bad-tests-v2/acc_on_device.f90
#include <openacc.h>

#ifndef T1
!T1:acc on device,V:2.7-2.3
      LOGICAL FUNCTION test1()
        USE OPENACC
        IMPLICIT NONE
        INCLUDE "acc_testsuite.Fh"
        
        INTEGER :: errors = 0

        ! Check if the code is running on the host
        IF (acc_on_device(acc_device_host) .EQV. .TRUE.) THEN
          errors = errors + 1
        END IF

        ! Check if the code is running on a device that is not the host
        IF (acc_on_device(acc_device_not_host) .EQV. .FALSE.) THEN
          errors = errors + 1
        END IF

        ! Check if the code is running on a specific device
        IF (acc_on_device(acc_device_type_nvidia) .EQV. .TRUE.) THEN
          errors = errors + 1
        END IF

        IF (errors .eq. 0) THEN
          test1 = .FALSE.
        ELSE
          test1 = .TRUE.
        END IF
      END
#endif

      PROGRAM main
        IMPLICIT NONE
        INTEGER :: failcode, testrun
        LOGICAL :: failed
        INCLUDE "acc_testsuite.Fh"
#ifndef T1
        LOGICAL :: test1
#endif
        failed = .FALSE.
        failcode = 0
#ifndef T1
        DO testrun = 1, NUM_TEST_CALLS
          failed = failed .or. test1()
        END DO
        IF (failed) THEN
          failcode = failcode + 2 ** 0
          failed = .FALSE.
        END IF
#endif
        CALL EXIT (failcode)
      END PROGRAM

File: Phind-CodeLlama-34B-v2/RAG-template/acc_on_device.c
#include "acc_testsuite.h"
#ifndef T1
//T1:acc on device,V:2.7-3.3
int test1(){
    int err = 0;

    // Check if the device is the host
    if (acc_on_device(acc_device_host) != 1) {
        err = 1;
    }

    // Check if the device is not the host
    if (acc_on_device(acc_device_not_host) != 0) {
        err = 1;
    }

    // Check if the device is an accelerator
    if (acc_on_device(acc_device_default) != 0) {
        err = 1;
    }

    return err;
}
#endif

int main(){
    int failcode = 0;
    int failed;
#ifndef T1
    failed = 0;
    for (int x = 0; x < NUM_TEST_CALLS; ++x){
        failed = failed + test1();
    }
    if (failed != 0){
        failcode = failcode + (1 << 0);
    }
#endif
    return failcode;
}